doc/source/changes/version_0_25_1.rst.inc

```diff
@@ -0,0 +1,33 @@
+New features
+------------
+
+* added a feature (see the :ref:`miscellaneous section <misc>` for details).
+
+* added another feature.
+
+.. _misc:
+
+Miscellaneous improvements
+--------------------------
+
+* setting values of a subset using another LArray is allowed only if axes and labels are compatible.
+  For example:
+
+    >>> arr = ndtest(4)
+    >>> arr
+    a  a0  a1  a2  a3
+        0   1   2   3
+    >>> arr['a2:'] = arr[':a1']
+    ...
+    ValueError: incompatible axes:
+    Axis(['a2', 'a3'], 'a')
+    vs
+    Axis(['a0', 'a1'], 'a')
+
+  is not allowed and raises a ValueError (closes :issue:`269`).
```

Comment:
* I consider not failing when it should have a (grave) bug (i.e. I'd move this to the fixes section). It is probably not obvious that it is an issue in this particular case (*) and I don't know how to make it obvious, but I have seen several occurrences of this silently corrupting user data. (*) and this is the reason I initially allowed this to work on purpose. 
* mentioning that it raises a ValueError does not seem useful (we see it just a few lines before). I would move the "issue closed mention" on the first line and kill this line entirely.
* I usually strip the traceback completely (i.e. remove even the "...")
* I think it might help users if you also mention how to circumvent this check (using drop_labels('a') or drop_labels() seems to be the cleanest way).